t/t1200-stats-basic.t

```diff
@@ -0,0 +1,92 @@
+#!/bin/sh
+#
+
+test_description='Test stats collection and sending'
+
+. `dirname $0`/sharness.sh
+
+test_under_flux 2
+
+export FLUX_FRIPP_STATSD=0.0.0.0:9999
+
+test_expect_success 'create udp client program' '
+	cat <<-EOF >listen.py &&
+	import argparse
+	import re
+	import socket
+	import time
+
+	parser = argparse.ArgumentParser()
+	parser.add_argument("-s", "--search-for", metavar="METRIC",
+						help="search for a specific metric tag. Must be used along with the timeout arg")
+	parser.add_argument("-t", "--timeout", help="timeout in s", type=float, default=10.0)
+	parser.add_argument("-V", "--validate", help="validate packet form", action="store_true")
+	parser.add_argument("-w", "--wait-for", metavar="N",
+						help="wait for N packets to be recieved. Should be used along with the timeout arg",
+						type=int, default=1)
+	args = parser.parse_args()
+
+	s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+	s.bind(("0.0.0.0", 9999))
+	s.settimeout(args.timeout)
```

Comment:
Instead of trying to internally track the timeout within `listen.py` it may be easier to run the program under our builtin `run_timeout` function. Then you don't have to worry about  tracking an overall timeout with blocking network code, which is difficult.

You could still keep the `-w, --wait-for=N` and `-s, --search-for=METRIC` options, just exit with zero status after N packets are received or you find the matching metric. On timeout, `run_timeout` will kill the script and trigger a failure.